src/mlir/consumer.h

```diff
@@ -86,4 +86,25 @@ namespace mlir::verona
     static llvm::Expected<OwningModuleRef>
     lower(MLIRContext* context, ::verona::parser::Ast ast);
   };
+
+  /*
+   * Scope Cleanup helper
+   *
+   * Automatically performs cleanup on destruction.
```

Comment:
```suggestion
   * Automatically invokes the callable object passed to the constructor on destruction.
   * This class is intended to provide lexically scoped cleanups, for example:
   * ```c++
   * ScopedCleanup defer([&] { /* cleanup code here */});
   * ```
   * The code in the lambda will be invoked when `defer` goes out of scope.
```